### PR TITLE
Convert migrate tests to integration tests

### PR DESCRIPTION
**Pull request by claudiu-cristea**

### CODE REVIEW

**Comment by claudiu-cristea** [comment]
This is blocked on #4835

**Comment by weitzman** [comment]
This is now unblocked. I changed the PR base branch to 11.x.

**Comment by claudiu-cristea** [comment]
This ready for review.

**IMPORTANT**: I had to raise the `memory_limit` to 256M in order to fix the PHP 8.1 test. But it's not clear for me if this is the correct way.

Other aspect: It would be nice not to install the testing modules on each test but only once, per class, using `::setUpBeforeClass()`. However, that is a static method, so it would add additional work. Not sure is worth it.

**Review by weitzman — APPROVED** [review]
Memory bump looks good to me.

I would like to reduce all those module enable/uninstall. I also cant think of a way, so lets just open a follow-up for that.

File: tests/integration/MigrateRunnerTest.php [hunk 1]
```diff
@@ -453,9 +451,10 @@ public function testCommandProgressBar(): void
         $this->assertNoProgressBar();
 
         // Set the 'test_migration' source plugin to skip count.
-        // @see woot_migrate_source_info_alter()
-        $this->drush('state:set', ['woot.test_command_progress_bar.skip_count', true]);
-        $this->drush('cache:rebuild');
+        // @see woot_migration_plugins_alter()
+        $this->drush('state:set', ['woot.migrate_runner.progress_bar_skip_count', true]);
+        // Get fresh plugin definitions.
+        drupal_flush_all_caches();
```
[the remark this answers is not in the document]
Comment (claudiu-cristea) [reply tests/integration/MigrateRunnerTest.php:457]: @weitzman for some reasons that doesn't work, I've tried before. I suspect that is a static cache issue. Also the old `drush cr` produced (no idea why):

> You have requested a non-existent service "drush.console.services".

File: tests/integration/MigrateRunnerTest.php [hunk 2]
```diff
@@ -519,4 +518,32 @@ protected function progressBarAssertionHelper(bool $assertHasProgressBar): void
...
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function tearDown(): void
+    {
+        // Cleanup any created content.
+        $storage = \Drupal::entityTypeManager()->getStorage('node');
+        $storage->delete($storage->loadMultiple());
```
[the remark this answers is not in the document]
Comment (claudiu-cristea) [reply tests/integration/MigrateRunnerTest.php]: This has revealed a bug in `11.x`

**Pull request merged**